### PR TITLE
Cleanup vrecord

### PR DESCRIPTION
**Pull request by aeschweik**
Standardize variables and functions, group functions and variable assignments, order for easier finding/editing, fix typos, more commenting.

### CODE REVIEW

File: vrecord [hunk 1]
```diff
@@ -4,24 +4,24 @@
 # Open-source software for capturing a video signal and turning it into a digital file.
 
 SCRIPTDIR=$(dirname "${0}")
-config_file="$HOME/.$(basename "${0}").conf"
-unset version
+CONFIG_FILE="$HOME/.$(basename "${0}").conf"
```
Comment (retokromer) [inline vrecord]: You could also use ${HOME} here.

File: vrecord [hunk 2]
```diff
@@ -61,13 +61,39 @@ Usage: $(basename "${0}") [ -g | -e | -r | -p | -a | -x | -h ]
...
+
+if [[ -f "${CONFIG_FILE}" ]] ; then
+    . "${CONFIG_FILE}"
+elif [[ "${RUNTYPE}" = "record" || "${RUNTYPE}" = "edit" ]] ; then
+    echo "No configuration file, restarting in edit mode."
+    touch "${CONFIG_FILE}"
+    exec "$(basename "${0}")" -e
+fi
+
+
```
Comment (retokromer) [inline vrecord]: Two empty lines?

File: vrecord [hunk 3]
```diff
@@ -109,215 +134,140 @@ _get_decklink_inputs(){
...
-    exec "$(basename "${0}")" -e
-fi
+    # Find Pashua binary. We do search both . and dirname "$0"
+    # , as in a doubleclickable application, cwd is /
+    # BTW, all these quotes below are necessary to handle paths
+    # containing spaces.
+    BUNDLEPATH="Pashua.app/Contents/MacOS/Pashua"
+    MYPATH=$(dirname "$0")
+    for searchpath in "$MYPATH/Pashua" "${MYPATH}/${BUNDLEPATH}" "./${BUNDLEPATH}" \
+                      "/Applications/${BUNDLEPATH}" "$HOME/Applications/${BUNDLEPATH}"
```
Comment (retokromer) [inline vrecord]: `${HOME}`?

File: vrecord [hunk 4]
```diff
@@ -109,215 +134,140 @@ _get_decklink_inputs(){
...
+    do
+        if [ -f "${searchpath}" -a -x "${searchpath}" ] ; then
+            PASHUAPATH="${searchpath}"
```
Comment (retokromer) [inline vrecord]: `SEARCHPATH`?

File: vrecord [hunk 5]
```diff
@@ -109,215 +134,140 @@ _get_decklink_inputs(){
...
+            fi
+        fi
+    else
+        # Get result
+        result=$("${PASHUAPATH}" "${PASHUA_CONFIGFILE}" | sed 's/ /;;;/g')
+
+        # Parse result
+        for line in $result ; do
+            key=`echo $line | sed 's/^\([^=]*\)=.*$/\1/'`
+            value=`echo $line | sed 's/^[^=]*=\(.*\)$/\1/' | sed 's/;;;/ /g'`
```
Comment (retokromer) [inline vrecord]: I suggest to use here as well `$( . . . )`

File: vrecord [hunk 6]
```diff
@@ -327,174 +277,268 @@ _writeingestlog(){
...
+
+_lookup_audio_mapping(){
     case "${1}" in
         "2 Stereo Tracks (Channels 1 & 2 -> 1st Track Stereo, Channels 3 & 4 -> 2nd Track Stereo)")
-            audiomap="[0:a:0]pan=stereo| c0=c0 | c1="$phase_value"c1[stereo1];[0:a:0]pan=stereo| c0=c2 | c1=c3[stereo2]"
-            map1k="-map"
-            map1v="[stereo1]"
-            map2k="-map"
-            map2v="[stereo2]"
+            AUDIOMAP="[0:a:0]pan=stereo| c0=c0 | c1="$PHASE_VALUE"c1[stereo1];[0:a:0]pan=stereo| c0=c2 | c1=c3[stereo2]"
```
Comment (retokromer) [inline vrecord]: `${PHASE_VALUE}`?

**Comment by aeschweik** [comment]
Thanks for your edits, @retokromer !

**Comment by dericed** [comment]
requires resolving conflicts

**Pull request merged**